src/cpp/fastrtps_deprecated/qos/ParameterList.cpp

```diff
@@ -689,11 +689,16 @@ bool ParameterList::readParameterListfromCDRMsg(CDRMessage_t& msg, std::function
                         p.m_value.push_back(static_cast<DataRepresentationId_t>(temp));
 
                     }
+                    if (size % 2 == 1) // Odd, we must read the alignment
+                    {
+                        valid &= CDRMessage::readInt16(&msg, &temp);
+                        std::cout << "Leido odd: " << temp << std::endl;
```

Comment:
What's this change? And also changes below from `uint16_t` to `uint32_t`?

Reply:
These changes were applied to fix serialization issues found by @richiware.